### PR TITLE
Building dotty artifact

### PR DESCRIPTION
**Pull request by LGLO**
Building for dotty.
Had to sacrifice better-monadic-for, because it's not released for dotty.
Added some ifs in build definition.

We would like to use it in https://github.com/softwaremill/sttp

### CODE REVIEW

File: project/BuildHelper.scala [hunk 1]
```diff
@@ -72,13 +73,19 @@ object BuildHelper {
...
-    libraryDependencies ++= compileOnlyDeps ++ Seq(
-      compilerPlugin("org.typelevel"   %% "kind-projector"  % "0.10.3"),
-      compilerPlugin("com.github.ghik" %% "silencer-plugin" % "1.4.2")
-    ),
+    libraryDependencies ++= {
+      if (isDotty.value)
+        Seq(("com.github.ghik" % "silencer-lib_2.13.1" % "1.6.0" % Provided).withDottyCompat(scalaVersion.value))
+      else
+        Seq(
+          "com.github.ghik" %% "silencer-lib" % "1.4.2" % "provided",
```
Comment (neko-kai) [inline project/BuildHelper.scala:84]: @LGLO You can get rid of `silencer-lib` by updating to `silencer-1.7.0` & using `scala.annotation.nowarn` (You'll need to add `scala-collection-compat` for `@nowarn` on 2.12 & 2.11)
https://github.com/ghik/silencer/releases/tag/v1.7.0

Comment (runtologist) [reply project/BuildHelper.scala:84]: I'm not sure if it is still needed at all, probably you can just remove it.

**Comment by runtologist** [comment]
Looks good!

File: project/BuildHelper.scala (hunk 1 above)
Comment (LGLO) [reply project/BuildHelper.scala]: Removed.
I also updated Scala versions and kind-projector version.

**Comment by runtologist** [comment]
awesome, LGTM. Thanks a lot.

**Pull request merged**